### DISCUSSION TITLE
micropython/lora can not run the async example

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#13266] [Libraries & Drivers] [answered]
Trying to run this example:
https://github.com/micropython/micropython-lib/tree/master/micropython/lora#asynchronous-api
The script will run once to the end of `async def recv_coro():` and stop receiving.  The `async def send_coro():` just keeps running.

I was having similar trouble running a `sync` version and the only way I can keep that and the async version running is to do a `machine.reset()` at the end of the `recv_coro()`.

Has anyone run that example successfully and hopefully be able suggest what basic step I have missed out.

Thanks

**Comment by sosi-deadeye** [comment, marked as answer]
The code in `recv_coro` runs only once. There is no loop.

Change

```python
async def recv_coro():
    rx = await modem.recv(2000)
    if rx:
        print(f'Received: {rx}')
    else:
        print('Timeout!')
```

to

```python
async def recv_coro():
    while True:
        rx = await modem.recv(2000):
        if rx:
            print(f'Received: {rx}')
        else:
            print('Timeout!')
```

**Reply by davefes** [reply]
Thanks ... one of my first attempts at async ... I thought the `async def init():`  just kept them both running :(

https://github.com/micropython/micropython-lib/tree/master/micropython/lora/examples

Unfortunately, I was hoping for "reliable_delivery" and "power-backoff" but it seems this async example is more like a "simple_rxtx"
